Issue upgrading to surefire version M5

In order to run some tests that are being changed in #2545, it appears that we need to upgrade the surefire version from M4 to M5. The issue with this is it causes certain tests to fail with hard to debug issues.
https://github.com/apache/accumulo/blob/8c07e0c8281fef857eeda03f78e287a37cdc6e47/pom.xml#L158-L159

At least the two tests mentioned above, [`RowHashIT.test`](https://github.com/apache/accumulo/blob/26821c1926a3b48943f3231725f8e2e0735d05fb/hadoop-mapreduce/src/test/java/org/apache/accumulo/hadoop/its/mapreduce/RowHashIT.java#L80) and [`ShellServerIT.scansWithClassLoaderContext`](https://github.com/apache/accumulo/blob/f442f94b1b2505d9425bbbcc940f26225bfe3930/test/src/main/java/org/apache/accumulo/test/shell/ShellServerIT.java#L1692) still fail when surefire is upgraded.

I wasn't sure if a separate issue should be made for these failing tests given that they do not fail on main and only fail when surefire is upgraded. Below is the strack trace and a few details for each test in case it might be evident to someone what a solution might be. I have not been able to make any progress in tracking down what the specific cause of the test failures might be and this is holding up progress on #2545.

<details>
  <summary>RowHashIT</summary>

```
[INFO] Running org.apache.accumulo.hadoop.its.mapreduce.RowHashIT
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 17.04 s <<< FAILURE! - in org.apache.accumulo.hadoop.its.mapreduce.RowHashIT
[ERROR] org.apache.accumulo.hadoop.its.mapreduce.RowHashIT.test  Time elapsed: 17.029 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at accumulo.hadoop.mapreduce@2.1.0-SNAPSHOT/org.apache.accumulo.hadoop.its.mapreduce.RowHashIT.runTest(RowHashIT.java:100)
	at accumulo.hadoop.mapreduce@2.1.0-SNAPSHOT/org.apache.accumulo.hadoop.its.mapreduce.RowHashIT.test(RowHashIT.java:82)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

</details>

<details>
  <summary>ShellServerIT</summary>

```
[INFO] Running org.apache.accumulo.test.shell.ShellServerIT
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.916 s <<< FAILURE! - in org.apache.accumulo.test.shell.ShellServerIT
[ERROR] org.apache.accumulo.test.shell.ShellServerIT.scansWithClassLoaderContext  Time elapsed: 4.115 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at accumulo.test@2.1.0-SNAPSHOT/org.apache.accumulo.test.shell.ShellServerIT.scansWithClassLoaderContext(ShellServerIT.java:1718)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)


```

</details>

I don't know about ShellServerIT, but I looked into RowHashIT, and ended up filing a bug upstream at [SUREFIRE-2035](https://issues.apache.org/jira/browse/SUREFIRE-2035) with an explanation of why I think this is happening.

RowHashIT launches a new process using `RowHash.main()` and the current test's class path. Here's the stack trace in the launched process that shows it is unable to find the classes that are in the main jar artifact for this module... something that should definitely be available on the test class path:

```java
[main] WARN  org.apache.accumulo.start.classloader.AccumuloClassLoader [] - 'general.classpaths' is deprecated but was set to '/home/christopher/git/apache/accumulo/accumulo/hadoop-mapreduce/target/mini-tests/org.apache.accumulo.hadoop.its.mapreduce.RowHashIT_test/lib/[^.].*[.]jar' 
[org.apache.accumulo.hadoop.its.mapreduce.RowHashIT$RowHash] ERROR org.apache.accumulo.start.Main [] - Thread 'org.apache.accumulo.hadoop.its.mapreduce.RowHashIT$RowHash' died.
java.lang.NoClassDefFoundError: org/apache/accumulo/hadoopImpl/mapreduce/lib/MapReduceClientOpts       
  at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]                                          
  at java.lang.ClassLoader.defineClass(ClassLoader.java:1017) ~[?:?]                                   
  at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174) ~[?:?]                    
  at jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800) ~[?:?]            
  at jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698) ~[?:?]
  at jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621) ~[?:?]        
  at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579) ~[?:?]              
  at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]           
  at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]                                      
  at org.apache.accumulo.hadoop.its.mapreduce.RowHashIT$RowHash.run(RowHashIT.java:145) ~[test-classes/:?]
  at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76) ~[hadoop-client-api-3.3.0.jar:?]        
  at org.apache.accumulo.hadoop.its.mapreduce.RowHashIT$RowHash.main(RowHashIT.java:172) ~[test-classes/:?]
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]                       
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]  
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
  at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]                                           
  at org.apache.accumulo.start.Main.lambda$execMainClass$1(Main.java:163) ~[accumulo-start-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
  at java.lang.Thread.run(Thread.java:829) ~[?:?]                                                      
Caused by: java.lang.ClassNotFoundException: org.apache.accumulo.hadoopImpl.mapreduce.lib.MapReduceClientOpts
  at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]              
  at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]           
  at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]                                      
  ... 18 more   
```

I set up a few strategic `System.err.println(System.getProperty("java.class.path"));` lines and was able to confirm that `accumulo-hadoop-mapreduce-2.1.0-SNAPSHOT.jar` is on the class path for 3.0.0-M4, but not for 3.0.0-M5. I have no idea why, or even how the tests still work without the main jar on the class path. I think maven-failsafe-plugin is doing something funny with its own class loader to filter out the main jar artifact for the module and provide those classes through its own class loader... or something crazy like that.

Just to add more detail to the ShellServerIT failure: 

<details>

```
[ERROR] org.apache.accumulo.test.shell.ShellServerIT.scansWithClassLoaderContext  Time elapsed: 1.865 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:35)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:179)
	at accumulo.test@2.1.0-SNAPSHOT/org.apache.accumulo.test.shell.ShellServerIT.scansWithClassLoaderContext(ShellServerIT.java:1715)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeout.lambda$assertTimeoutPreemptively$4(AssertTimeout.java:138)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```
</details>

The line correlates to an assert to verify that the shell iterator class does not `implement OptionDescriber`.

```
result = ts.exec("setshelliter -pn baz -n reverse -p 21 -class " + VALUE_REVERSING_ITERATOR);
 assertTrue(result.contains("The iterator class does not implement OptionDescriber"));
```

--- a/pom.xml
+++ b/pom.xml
@@ -155,8 +155,7 @@
     <surefire.forkCount>1C</surefire.forkCount>
     <surefire.groups />
     <surefire.reuseForks>true</surefire.reuseForks>
-    <!-- 3.0.0-M5 causes RowHashIT.test and ShellServerIT.scansWithClassLoaderContext to fail -->
-    <surefire.version>3.0.0-M4</surefire.version>
+    <surefire.version>3.0.0-M6</surefire.version>
     <!-- Thrift version -->
     <thrift.version>0.15.0</thrift.version>
     <unitTestMemSize>-Xmx1G</unitTestMemSize>

